test/cpp/end2end/async_end2end_test.cc

```diff
@@ -67,10 +67,10 @@ namespace {
 void* tag(int i) { return (void*)(intptr_t)i; }
 
 #ifdef GPR_POSIX_SOCKET
-static int assert_non_blocking_poll(struct pollfd* pfds, nfds_t nfds,
+static int non_blocking_poll(struct pollfd* pfds, nfds_t nfds,
                                     int timeout) {
-  GPR_ASSERT(timeout == 0);
-  return poll(pfds, nfds, timeout);
+  /* ignore timeout and always use timeout 0 */
```

Comment:
Removing the assert makes this a much weaker test.

The original test provided guarantees that we never blocked calling code, which is important for many use cases (imagine a UI thread that wants to do *some* grpc work).

Overriding that here means that we're no longer providing such a guarantee.